### PR TITLE
Add Queued State to Analysis State

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
BRICR fixes.

#### What's this PR do?
* Add 'queued' state to property.analysis_state
* Fix the update methods to use the PropertyStateSerializer and TaxLotStateSerializer. This fixed the issue with the enumeration string being passed into the API.

#### How should this be manually tested?
* BRICR tests, monkey tests, unit tests

#### What are the relevant tickets?
N/A

#### Definition of Done:
- [x] Is there appropriate test coverage? (e.g. ChefSpec, Mocha/Chai, Python, etc.)
- [x] Does this PR require a Selenium test? (e.g. Browser-specific bugs or complicated UI bugs)
- [x] Does this PR require a regression test? All fixes require a regression test.
- [x] Does this add new dependencies? If so, does PIP, npm, bower requirements need to be updated?

### CODE REVIEW

File: seed/api/v2_1/views.py [hunk 1]
```diff
@@ -251,6 +253,36 @@ def hpxml(self, request, pk):
...
+        # # Move the old building file to the new state to perserve the history
+        # for b in old_state.building_files.all():
+        #     b.property_state = new_state
+        #     b.save()
+        #
+        # for s in Simulation.objects.filter(property_state=old_state):
+        #     s.property_state = new_state
+        #     s.save()
+
+        return new_state
```
Comment (Myoldmopar) [inline seed/api/v2_1/views.py]: This code was intentionally commented out, right?

File: seed/api/v2_1/views.py [hunk 2]
```diff
@@ -311,12 +343,16 @@ def update_with_building_sync(self, request, pk):
...
-
+            previous_state = property_view.state
             # passing in the existing propertyview allows it to process the buildingsync file and attach it to the
             # existing propertyview.
             p_status, new_pv_state, new_pv_view, messages = building_file.process(
                 organization_id, cycle, property_view=property_view
             )
+
+            # merge the relationships from the old property state
+            self._merge_relationships(previous_state, new_pv_state)
```
Comment (Myoldmopar) [inline seed/api/v2_1/views.py]: This has no side effects, at the moment.  It looks like it is just all stubbed out for later.  In which case, OK.

File: seed/views/properties.py [hunk 3]
```diff
@@ -642,12 +646,11 @@ def update(self, request, pk=None):
                 ).order_by('-id').first()
 
                 if 'extra_data' in new_property_state_data.keys():
-                    property_state_data['extra_data'].update(
-                        new_property_state_data.pop('extra_data'))
+                    property_state_data['extra_data'].update(new_property_state_data.pop('extra_data'))
                 property_state_data.update(new_property_state_data)
 
                 if log.name == 'Import Creation':
```
Comment (Myoldmopar) [inline seed/views/properties.py]: These strings in the if blocks seem like voodoo, but it's not like you added them here, so OK.

File: seed/views/properties.py [hunk 4]
```diff
@@ -695,20 +691,30 @@ def update(self, request, pk=None):
...
-                    result.update(
-                        {'state': PropertyStateSerializer(state).data}
+                    # Convert this to using the serializer to save the data. This will override the previous values
+                    # in the state object.
+
+                    # Note: We should be able to use partial update here and pass in the changed fields instead of the
+                    # entire state_data.
+                    updated_property_state_serializer = PropertyStateSerializer(
+                        property_view.state,
+                        data=property_state_data
```
Comment (Myoldmopar) [inline seed/views/properties.py]: So much nicer!

File: seed/api/v2_1/views.py (hunk 1 above)
Comment (nllong) [reply seed/api/v2_1/views.py]: yes

File: seed/api/v2_1/views.py (hunk 2 above)
Comment (nllong) [reply seed/api/v2_1/views.py]: yes, it has to be hooked up when someone can think about how we merge measures and scenarios.

File: seed/static/seed/partials/inventory_detail.html [hunk 5]
```diff
@@ -281,7 +281,7 @@ <h2><i class="fa fa-cogs"></i> Measures</h2>
                     <div class="section_header fixed_height_short has_no_padding">
                         <div class="section_action_container left" style="width: 50%;">
                         <span>
-                            <h2><i class="fa fa-rebel"></i> Scenarios</h2>
+                            <h2><i class="fa fa-sitemap"></i> Scenarios</h2>
```
[the remark this answers is not in the document]
Comment (nllong) [reply seed/static/seed/partials/inventory_detail.html]: I know... had to grow up.

File: seed/views/properties.py (hunk 4 above)
Comment (nllong) [reply seed/views/properties.py]: right?!

**Comment by Myoldmopar** [comment]
Do you need a merger or will you be doing it @nllong ?

**Comment by nllong** [comment]
Done

**Pull request merged**